src/lib/useEmojis.ts

```diff
@@ -9,20 +9,29 @@ import type {
 
 function reducer(state: EmojiiState, action: EmojiAction) {
   const { emoji } = action;
-  const emojiFromState = state.find((em) => em.emoji === emoji.emoji) || emoji;
+  const stateEmoji = state.find((em) => em.emoji === emoji.emoji);
+  const emojiFromState = stateEmoji || emoji;
   emojiFromState.counter = emojiFromState.counter || 0;
   switch (action.type) {
     case 'i':
       emojiFromState.counter++;
       break;
     case 'd':
       emojiFromState.counter--;
-      if (emojiFromState.counter < 0) emojiFromState.counter = 0;
+      if (emojiFromState.counter <= 0) emojiFromState.counter = 0;
   }
-  return [
-    ...state.filter((rea) => rea.emoji !== emojiFromState.emoji),
-    ...(emojiFromState.counter > 0 ? [emojiFromState] : []),
-  ];
+  if (!stateEmoji) {
+    state.push(emojiFromState);
+  }
+
+  return state
```

Comment:
That's a nice way to keep the state static! Nice work. I just changed this return a bit to make it more understandable (I hope) and to reduce the bundle size. You can see this change in the next commit, check it out and tell me if it looks good to you.
Before:   
```
Wrote 909 B: index.js.gz
        788 B: index.js.br
Wrote 832 B: index.modern.js.gz
        723 B: index.modern.js.br
```

After:  
```
Wrote 827 B: index.modern.js.gz
        722 B: index.modern.js.br
Wrote 902 B: index.js.gz
        790 B: index.js.br
```

Reply:
nice. please push and I will look into it

Reply:
looks great